torch/testing/_internal/common_methods_invocations.py

```diff
@@ -16852,7 +16855,87 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
     ),
 ]
 
+# NOTE [Python References]
+# Python References emulate existing PyTorch operations, but can ultimately
+#   be expressed in terms of "primitive" operations from torch._prims.
+#
+# These references are experimental.
+# See https://dev-discuss.pytorch.org/t/tracing-with-primitives-update-0/577
+#   for additional context.
+#
+# Python Reference OpInfos should be added to the python_ref_db list below.
+#   Tests can opt-into running on these references by including
+#   that list in the Sequence they pass to the @ops decorator.
+
+class ElementwiseBinaryPythonRefInfo(BinaryUfuncInfo):
+    '''
+    An OpInfo for a Python reference to an elementwise binary operation.
+
+    When constructing this OpInfo a pointer to an existing OpInfo must be
+    provided using the torch_opinfo_name kwarg. An OpInfo with the same
+    name as that string (and no variant) will be found to inherit from.
+
+    Instead of just inheriting the existing OpInfo's metadata, this
+    will actually reconstruct the appropriate OpInfo metadata by
+    inheriting the existing OpInfo's initialization arguments. These
+    arguments can be overridden by adding kwargs to the constructor.
+    '''
+    def __init__(
+            self,
+            name,  # the stringname of the callable Python reference
+            *,
+            op=None,  # the function variant of the operation, populated as torch.<name> if None
+            torch_opinfo_name,  # the string name of the corresponding torch opinfo
+            **kwargs):  # additional kwargs override kwargs inherited from the torch opinfo
+
+        # TODO: extract this into a common helper
+        self.torch_opinfo_name = torch_opinfo_name
+
+        # finds the corresponding opinfo
+        self.torch_opinfo = None
+        for opinfo in op_db:
+            if opinfo.name == torch_opinfo_name and opinfo.variant_test_name == '':
```

Comment:
That requires uncommon self-control from OpInfo writers to not put in random variant names, and how do we know that the one without variant name is the best?

Reply:
We don't and in the future we may have to create a better system. But we'll assert if things go too wrong